app/models/school_device_allocation.rb

```diff
@@ -21,6 +21,22 @@ def self.by_device_type(device_type)
     where(device_type: device_type)
   end
 
+  def self.has_fully_ordered
```

Comment:
can we convert these class methods to scopes?

Reply:
Yes, we are quite inconsistent in this, good shout.

Reply:
Done.